docx/oxml/document.py

```diff
@@ -57,3 +59,13 @@ def clear_content(self):
             content_elms = self[:]
         for content_elm in content_elms:
             self.remove(content_elm)
+
+    @property
+    def sdts(self):
+        for sdt in self.sdt_lst:
+            yield sdt, sdt.name
```

Comment:
I think I was wrong in `r_lst_recursive` to make it property and generator at the same time. I think it is bad as property should be something that could be used not only in context of iteration. Furthermore, generators could be exhausted after iteration is done which changes semantics of expected collection at the place of the property. 
The same reasoning is applied here and to `sdts_all` bellow.